src/nwb_conversion_tools/tools/roiextractors/roiextractors.py

```diff
@@ -212,76 +213,85 @@ def get_nwb_imaging_metadata(imgextractor: ImagingExtractor):
         the sampling rate is used.
     """
     assert save_path is None or nwbfile is None, "Either pass a save_path location, or nwbfile object, but not both!"
-
     if nwbfile is not None:
         assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
-    # Update any previous metadata with user passed dictionary
+
+    # TODO on or after August 1st, 2022, remove argument and deprecation warnings
+    if save_path is not None:
+        will_be_removed_str = "will be removed on or after August 1st, 2022. Please use 'nwbfile_path' instead."
+        if nwbfile_path is not None:
+            if save_path == nwbfile_path:
+                warn(
+                    "Passed both 'save_path' and 'nwbfile_path', but both are equivalent! "
+                    f"'save_path' {will_be_removed_str}",
+                    DeprecationWarning,
+                )
+            else:
+                warn(
+                    "Passed both 'save_path' and 'nwbfile_path' - using only the 'nwbfile_path'! "
+                    f"'save_path' {will_be_removed_str}",
+                    DeprecationWarning,
+                )
+        else:
+            warn(
+                f"The keyword argument 'save_path' to 'spikeinterface.write_recording' {will_be_removed_str}",
+                DeprecationWarning,
+            )
+            nwbfile_path = save_path
+
     if metadata is None:
         metadata = dict()
     if hasattr(imaging, "nwb_metadata"):
         metadata = dict_deep_update(imaging.nwb_metadata, metadata)
     metadata = dict_deep_update(get_nwb_imaging_metadata(imaging), metadata)
-    if nwbfile is None:
-        save_path = Path(save_path)
-        assert save_path.suffix == ".nwb", "'save_path' file is not an .nwb file"
-
-        if save_path.is_file():
-            if not overwrite:
-                read_mode = "r+"
-            else:
-                # save_path.unlink()
-                read_mode = "w"
-        else:
-            read_mode = "w"
-        with NWBHDF5IO(str(save_path), mode=read_mode) as io:
-            if read_mode == "r+":
-                nwbfile = io.read()
-            else:
-                nwbfile = make_nwbfile_from_metadata(metadata=metadata)
-                add_devices(nwbfile=nwbfile, metadata=metadata)
-                add_two_photon_series(
-                    imaging=imaging,
-                    nwbfile=nwbfile,
-                    metadata=metadata,
-                    buffer_size=buffer_size,
-                )
-                add_epochs(imaging=imaging, nwbfile=nwbfile)
-            io.write(nwbfile)
-    else:
+    with make_or_load_nwbfile(
+        nwbfile_path=nwbfile_path, nwbfile=nwbfile, metadata=metadata, overwrite=overwrite, verbose=verbose
+    ) as nwbfile_out:
         add_devices(nwbfile=nwbfile, metadata=metadata)
-        add_two_photon_series(imaging=imaging, nwbfile=nwbfile, metadata=metadata, use_times=use_times)
```

Comment:
Why are you removing `use_times`? this argument is not used anymore it seems.

Reply:
Oops, good catch. Reverted.

Reply:
The tests did not catch this, though? I guess that's something we are missing.